src/Illuminate/Database/Eloquent/Concerns/HasPrimaryUlid.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Illuminate\Database\Eloquent\Concerns;
+
+use Illuminate\Support\Str;
+
+trait HasPrimaryUlid
+{
+    /**
+     * Generate a primary ULID for the model.
+     *
+     * @return void
+     */
+    public static function bootHasPrimaryUlid()
+    {
+        static::creating(function (self $model) {
+            if (empty($model->{$model->getKeyName()})) {
+                $model->{$model->getKeyName()} = $model->generatePrimaryUlid();
+            }
+        });
+    }
+
+    /**
+     * Generate the primary ULID key for the model.
+     *
+     * @return string
+     */
+    public function generatePrimaryUlid()
```

Comment:
I wonder if `generatePrimaryUlid()` and `generatePrimaryUuid()` should be generalized and renamed to both be `generatePrimaryKey()`. If accessing these methods manually, it will remove some friction if different models are using different variations of the uuid.

Reply:
Done.